doc/source/basics.rst

```diff
@@ -793,6 +793,79 @@ This is equivalent to the following
    result
    result.loc[:,:,'ItemA']
 
+.. _basics.assigning:
+
+Assigning New Columns
+---------------------
+
+.. versionadded:: 0.16.0
+
+Inspired by `dplyr's
+<http://cran.rstudio.com/web/packages/dplyr/vignettes/introduction.html#mutate>`__
+``mutate`` verb, DataFrame has an :meth:`~pandas.DataFrame.assign`
+method that allows you to easily create new columns that are potentially
+derived from existing columns. 
+
+.. ipython:: python
+
+   iris = read_csv('data/iris.data')
+   iris.head()
+
+   iris.assign(sepal_ratio=iris['SepalWidth'] / iris['SepalLength']).head()
+
+Above was an example of inserting a precomputed value. We can also pass in
+a function of one argument to be evalutated on the DataFrame being assigned to.
+
+.. ipython:: python
+
+   iris.assign(sepal_ratio = lambda x: x['SepalWidth'] /
```

Comment:
It would be best to pick a style here for spacing around `=` and stick to it for the docs. Personally, I like the extra spaces (even though it goes against the usual PEP8) because it's used for assignment, and it looks very unbalanced otherwise with the lambda.

Also: for these lines, I would consider using parentheses like this to group the lambda statement:
```
lambda x: (x['SepalWidth'] /
           x['SepalLength])
```
(just looks better to me)